doc/barclamp.md

```diff
@@ -0,0 +1,1321 @@
+  forked [crowbar-openstack](https://www.github.com/crowbar/crowbar-openstack)
+  throughout the rest of this document.
+
+* A Crowbar admin node with the Crowbar version you are developing against
+  (for new barclamps this will be `master`) already installed and running. This
+  should be as up-to-date as possible to reduce the chance of merge conflicts
+  happening.
+
+* A set of machines for your Crowbar admin node to orchestrate. In the case of
+  SUSE OpenStack Cloud this would be a controller and one compute node at a
+  minimum.
+
+* You will need to install the following packages:
+
+  * `patch`
+  * `git-core`
+
+Providing these is out of scope for this document. We assume you are able to
+fork the appropriate Barclamp collection. The machines can be provided through
+manual installation or by using an automated setup tool such as
+[mkcloud](https://github.com/SUSE-Cloud/automation/blob/master/scripts/mkcloud).
+Depending on the individual situation you can use either the `plain` or the
+`installonly` `mkcloud` step. The former will install the nodes and deploy the
+standard set of barclamps in their default configuration, while the latter will
+only install the nodes. Use `plain` if your barclamp does not need to be
+deployed before other barclamps and if you do not modify other barclamps.
+Otherwise use `installonly` and deploy barclamps bit by bit through the Crowbar
+web UI or the Crowbar CLI client. This will give you the opportunity to add
+package repositories or tweak barclamps' proposal settings where needed along
+the way.
+
+Optionally (but strongly recommended) you can also
+
+* Create a backup of your Crowbar admin node: You are probably going to break
+  it a few times in the course of testing your new barclamp so it is a good
+  idea to have a backup you can quickly restore rather than rebuild your entire
+  test environment. If you are using `mkcloud` you can create and restore a LVM
+  snapshot of your admin node using its `createadminsnapshot` and
+  `restoreadminfromsnapshot` steps, respectively.
+
+* Create an additional OpenStack controller or compute node as a scratch node:
+  this will allow you to deploy only your own barclamp's controller side
+  components to this node while all other OpenStack services reside on the main
+  controller. This greatly shortens the time a chef run for your barclamp's
+  controller side components takes and thus speeds up development. Make a
+  backup of this node as well to guard against breakage and allow for
+  clean-slate deployment of your barclamp's controller side components.
+
+The best way to set up your scratch node is to make it a lonely node. To this
+end, you will need to re-run mkcloud after you have created your cloud:
+
+```
+nodenumberlonelynode=1 mkcloud setuplonelynodes crowbar_register
+
+```
+
+## Development Environment
+
+Most of your development will take place on your Crowbar node (log in as root).
```

Comment:
Yes. In fact this is what I am counting on. It will provide a uniform environment where editor, configuration files, ssh keys (or the lack thereof are known quantities).